### PR TITLE
MM-42094 Rework handling of uncaught JS errors

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR reworks how the webapp handles uncaught JS errors.

Firstly, it removes the unnecessary request in the `onerror` handlers, which causes an 401 anyway, as it doesn't contain a CSRF token.

Secondly, all JS errors now get logged - on `debug` level. This should help support with debugging issues.

If developers mode is enabled, the announcement banner shows the error information. The error message was added back. The background color is back to red.

Example log message on the server:
```
debug [2022-02-25 11:25:13.147 +01:00] Client Logs API Endpoint Message: {"type":"developer","message":"A JavaScript error has occurred. Please use the JavaScript console to capture and report the error (msg: Uncaught foo, row: 37, col: 159561).","stack":"foo","url":"http://localhost:8065/static/plugins/com.mattermost.plugin-todo/com.mattermost.plugin-todo_2dcd215ae8d8470e_bundle.js"} caller="api4/system.go:371" type=client_message user_agent="Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36"
```

#### Screenshots
Screenshot from 2022-02-25 11-22-58

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-42094

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by mickmister — APPROVED** [review]
LGTM :tada:

File: root.jsx [hunk 1]
```diff
@@ -25,18 +26,22 @@ function preRenderSetup(callwhendone) {
...
+        let displayable = false;
         if (isDevMode()) {
-            store.dispatch(logError({type: 'developer', message: 'DEVELOPER MODE: A JavaScript error has occurred.  Please use the JavaScript console to capture and report the error (row: ' + line + ' col: ' + column + ').'}, true));
+            displayable = true;
         }
+
+        store.dispatch(
+            logError({
+                type: AnnouncementBarTypes.DEVELOPER,
+                message: 'A JavaScript error has occurred. Please use the JavaScript console to capture and report the error (msg: ' + msg + ', row: ' + line + ', col: ' + column + ').',
```
Comment (hmhealey) [inline root.jsx:38]: We could probably file this as a separate ticket, but maybe we could send this error to the server without the UI text? It seems weird for the logs on the server to include the "Please use the JS console..." part

File: packages/mattermost-redux/src/types/client4.ts [hunk 2]
```diff
@@ -1,8 +1,12 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-// I assume these are the log levels
-export type LogLevel = 'ERROR' | 'WARNING' | 'INFO';
+export enum LogLevel {
```
Comment (hmhealey) [inline packages/mattermost-redux/src/types/client4.ts:4]: I love seeing more enums. We need to do some more of this at some point to get rid of a lot of constants

**Review by catalintomai — APPROVED** [review]

File: root.jsx (hunk 1 above)
Comment (hanzei) [reply root.jsx]: Sure, please see cb9e074f1360c6e4fc54f970c29cf23fe7dca051.

**Review by hmhealey — APPROVED** [review]

**Pull request merged**